web/src/app/pages/main-page-container/main-page/map/map.component.ts

```diff
@@ -166,6 +175,76 @@ export class MapComponent implements AfterViewInit, OnDestroy {
         this.navigationService.getSubmissionId$(),
       ]).subscribe(() => this.cancelReposition())
     );
+
+    this.subscription.add(
+      this.submissionService.getSelectedSubmission$().subscribe(submission => {
```

Comment:
Would this cause a race condition with the `submissionId` and `locationOfInterestId` above?